Address more safer cpp warnings in WebKit/WebProcess/InjectedBundle

cdumez opened the pull request:
#### 592a0de98827ed52ded85a4eece56e8f6610dc46
<pre>
Address more safer cpp warnings in WebKit/WebProcess/InjectedBundle
<a href="https://bugs.webkit.org/show_bug.cgi?id=301839">https://bugs.webkit.org/show_bug.cgi?id=301839</a>

Reviewed by Darin Adler and Anne van Kesteren.

* Source/WebCore/rendering/HitTestResult.cpp:
(WebCore::HitTestResult::protectedInnerNonSharedNode const):
(WebCore::HitTestResult::protectedURLElement const):
* Source/WebCore/rendering/HitTestResult.h:
* Source/WebKit/SaferCPPExpectations/UncheckedCallArgsCheckerExpectations:
* Source/WebKit/SaferCPPExpectations/UncheckedLocalVarsCheckerExpectations:
* Source/WebKit/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
* Source/WebKit/SaferCPPExpectations/UncountedLocalVarsCheckerExpectations:
* Source/WebKit/WebProcess/InjectedBundle/API/Cocoa/WKWebProcessPlugInCSSStyleDeclarationHandle.mm:
(-[WKWebProcessPlugInCSSStyleDeclarationHandle dealloc]):
* Source/WebKit/WebProcess/InjectedBundle/API/Cocoa/WKWebProcessPlugInFrame.mm:
(protectedFrame):
(-[WKWebProcessPlugInFrame dealloc]):
(-[WKWebProcessPlugInFrame jsContextForWorld:]):
(-[WKWebProcessPlugInFrame jsContextForServiceWorkerWorld:]):
(-[WKWebProcessPlugInFrame hitTest:]):
(-[WKWebProcessPlugInFrame hitTest:options:]):
(-[WKWebProcessPlugInFrame jsCSSStyleDeclarationForCSSStyleDeclarationHandle:inWorld:]):
(-[WKWebProcessPlugInFrame jsNodeForNodeHandle:inWorld:]):
(-[WKWebProcessPlugInFrame jsRangeForRangeHandle:inWorld:]):
(-[WKWebProcessPlugInFrame _browserContextController]):
(-[WKWebProcessPlugInFrame URL]):
(-[WKWebProcessPlugInFrame childFrames]):
(-[WKWebProcessPlugInFrame containsAnyFormElements]):
(-[WKWebProcessPlugInFrame isMainFrame]):
(-[WKWebProcessPlugInFrame _securityOrigin]):
(-[WKWebProcessPlugInFrame appleTouchIconURLs]):
(-[WKWebProcessPlugInFrame faviconURLs]):
(-[WKWebProcessPlugInFrame _parentFrame]):
(-[WKWebProcessPlugInFrame _hasCustomContentProvider]):
(-[WKWebProcessPlugInFrame _certificateChain]):
(-[WKWebProcessPlugInFrame _serverTrust]):
(-[WKWebProcessPlugInFrame _provisionalURL]):
* Source/WebKit/WebProcess/InjectedBundle/API/Cocoa/WKWebProcessPlugInHitTestResult.mm:
(-[WKWebProcessPlugInHitTestResult nodeHandle]):
* Source/WebKit/WebProcess/InjectedBundle/API/Cocoa/WKWebProcessPlugInNodeHandle.mm:
(protectedNodeHandle):
(-[WKWebProcessPlugInNodeHandle dealloc]):
(-[WKWebProcessPlugInNodeHandle htmlIFrameElementContentFrame]):
(-[WKWebProcessPlugInNodeHandle renderedImageWithOptions:width:]):
(-[WKWebProcessPlugInNodeHandle elementBounds]):
(-[WKWebProcessPlugInNodeHandle HTMLInputElementIsAutoFilled]):
(-[WKWebProcessPlugInNodeHandle HTMLInputElementIsAutoFilledAndViewable]):
(-[WKWebProcessPlugInNodeHandle HTMLInputElementIsAutoFilledAndObscured]):
(-[WKWebProcessPlugInNodeHandle setHTMLInputElementIsAutoFilled:]):
(-[WKWebProcessPlugInNodeHandle setHTMLInputElementIsAutoFilledAndViewable:]):
(-[WKWebProcessPlugInNodeHandle setHTMLInputElementIsAutoFilledAndObscured:]):
(-[WKWebProcessPlugInNodeHandle isHTMLInputElementAutoFillButtonEnabled]):
(-[WKWebProcessPlugInNodeHandle setHTMLInputElementAutoFillButtonEnabledWithButtonType:]):
(-[WKWebProcessPlugInNodeHandle htmlInputElementAutoFillButtonType]):
(-[WKWebProcessPlugInNodeHandle htmlInputElementLastAutoFillButtonType]):
(-[WKWebProcessPlugInNodeHandle HTMLInputElementIsUserEdited]):
(-[WKWebProcessPlugInNodeHandle HTMLTextAreaElementIsUserEdited]):
(-[WKWebProcessPlugInNodeHandle isSelectElement]):
(-[WKWebProcessPlugInNodeHandle isSelectableTextNode]):
(-[WKWebProcessPlugInNodeHandle isTextField]):
(-[WKWebProcessPlugInNodeHandle HTMLTableCellElementCellAbove]):
(-[WKWebProcessPlugInNodeHandle frame]):
* Source/WebKit/WebProcess/InjectedBundle/API/Cocoa/WKWebProcessPlugInRangeHandle.mm:
(protectedRangeHandle):
(-[WKWebProcessPlugInRangeHandle dealloc]):
(-[WKWebProcessPlugInRangeHandle frame]):
(-[WKWebProcessPlugInRangeHandle text]):
* Source/WebKit/WebProcess/InjectedBundle/API/Cocoa/WKWebProcessPlugInScriptWorld.mm:
(protectedWorld):
(-[WKWebProcessPlugInScriptWorld dealloc]):
(-[WKWebProcessPlugInScriptWorld clearWrappers]):
(-[WKWebProcessPlugInScriptWorld makeAllShadowRootsOpen]):
(-[WKWebProcessPlugInScriptWorld exposeClosedShadowRootsForExtensions]):
(-[WKWebProcessPlugInScriptWorld disableOverrideBuiltinsBehavior]):
* Source/WebKit/WebProcess/InjectedBundle/API/mac/WKDOMDocument.mm:
(-[WKDOMDocumentParserYieldToken initWithDocument:]):
* Source/WebKit/WebProcess/InjectedBundle/API/mac/WKDOMInternals.h:
* Source/WebKit/WebProcess/InjectedBundle/API/mac/WKDOMNode.mm:
(-[WKDOMNode _initWithImpl:]):
* Source/WebKit/WebProcess/InjectedBundle/API/mac/WKDOMRange.mm:
(-[WKDOMRange _initWithImpl:]):
* Source/WebKit/WebProcess/InjectedBundle/InjectedBundleHitTestResult.cpp:
(WebKit::InjectedBundleHitTestResult::nodeHandle const):
(WebKit::InjectedBundleHitTestResult::urlElementHandle const):
(WebKit::InjectedBundleHitTestResult::imageRect const):

Canonical link: <a href="https://commits.webkit.org/302475@main">https://commits.webkit.org/302475@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b35909ceeaee6bef1b170cbef7c167232c2a0fba

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/129208 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/1466 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/40044 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/136585 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/80597 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/de31b8fc-c7d4-4034-80bf-d8ba8fbbddee) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/131079 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/1398 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/1343 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/98386 "Passed tests") | [❌ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/66284 "") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/e97516e3-4256-4c38-ab08-01d104de8fc3) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/132155 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/162/builds/1093 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/115741 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/79036 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/f7763f74-4648-455d-8906-8c91c57f5f31) 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/154/builds/1014 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/33859 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/79864 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/109464 "Passed tests") | [❌ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/34354 "Found 1 new test failure: http/tests/site-isolation/window-open-with-name-cross-site.html (failure)") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/139059 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/1258 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/1213 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/106918 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/1309 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/112077 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/106754 "Passed tests") | | 
| [  ~~🛠 🧪 merge~~](https://ews-build.webkit.org/#/builders/19/builds/27179 "The change is no longer eligible for processing.") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/1037 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/30601 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/53858 "Built successfully") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/20170 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/1331 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/64685 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/1156 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/1199 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/1255 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

annevk commented on Source/WebKit/WebProcess/InjectedBundle/API/Cocoa/WKWebProcessPlugInFrame.mm:
```diff
@@ -70,78 +77,79 @@ - (void)dealloc
 - (JSContext *)jsContextForWorld:(WKWebProcessPlugInScriptWorld *)world
 {
-    return [JSContext contextWithJSGlobalContextRef:_frame->jsContextForWorld(&[world _scriptWorld])];
+    return [JSContext contextWithJSGlobalContextRef:protectedFrame(self)->jsContextForWorld(Ref { [world _scriptWorld] }.ptr())];
```
It seems like there are enough instances for _protectedScriptWorld although I always get a bit confused by Objective-C.

cdumez replied:
It is defined in objective C like this:
```
@property (readonly) WebKit::InjectedBundleScriptWorld& _scriptWorld;
```

Not C++ so I left alone. I am not super familiar with ObjC. Maybe it's OK to add a second property `_protectedScriptWorld`, but I'm not 100% sure.

cdumez committed:
Address more safer cpp warnings in WebKit/WebProcess/InjectedBundle
https://bugs.webkit.org/show_bug.cgi?id=301839

Reviewed by Darin Adler and Anne van Kesteren.

* Source/WebCore/rendering/HitTestResult.cpp:
(WebCore::HitTestResult::protectedInnerNonSharedNode const):
(WebCore::HitTestResult::protectedURLElement const):
* Source/WebCore/rendering/HitTestResult.h:
* Source/WebKit/SaferCPPExpectations/UncheckedCallArgsCheckerExpectations:
* Source/WebKit/SaferCPPExpectations/UncheckedLocalVarsCheckerExpectations:
* Source/WebKit/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
* Source/WebKit/SaferCPPExpectations/UncountedLocalVarsCheckerExpectations:
* Source/WebKit/WebProcess/InjectedBundle/API/Cocoa/WKWebProcessPlugInCSSStyleDeclarationHandle.mm:
(-[WKWebProcessPlugInCSSStyleDeclarationHandle dealloc]):
* Source/WebKit/WebProcess/InjectedBundle/API/Cocoa/WKWebProcessPlugInFrame.mm:
(protectedFrame):
(-[WKWebProcessPlugInFrame dealloc]):
(-[WKWebProcessPlugInFrame jsContextForWorld:]):
(-[WKWebProcessPlugInFrame jsContextForServiceWorkerWorld:]):
(-[WKWebProcessPlugInFrame hitTest:]):
(-[WKWebProcessPlugInFrame hitTest:options:]):
(-[WKWebProcessPlugInFrame jsCSSStyleDeclarationForCSSStyleDeclarationHandle:inWorld:]):
(-[WKWebProcessPlugInFrame jsNodeForNodeHandle:inWorld:]):
(-[WKWebProcessPlugInFrame jsRangeForRangeHandle:inWorld:]):
(-[WKWebProcessPlugInFrame _browserContextController]):
(-[WKWebProcessPlugInFrame URL]):
(-[WKWebProcessPlugInFrame childFrames]):
(-[WKWebProcessPlugInFrame containsAnyFormElements]):
(-[WKWebProcessPlugInFrame isMainFrame]):
(-[WKWebProcessPlugInFrame _securityOrigin]):
(-[WKWebProcessPlugInFrame appleTouchIconURLs]):
(-[WKWebProcessPlugInFrame faviconURLs]):
(-[WKWebProcessPlugInFrame _parentFrame]):
(-[WKWebProcessPlugInFrame _hasCustomContentProvider]):
(-[WKWebProcessPlugInFrame _certificateChain]):
(-[WKWebProcessPlugInFrame _serverTrust]):
(-[WKWebProcessPlugInFrame _provisionalURL]):
* Source/WebKit/WebProcess/InjectedBundle/API/Cocoa/WKWebProcessPlugInHitTestResult.mm:
(-[WKWebProcessPlugInHitTestResult nodeHandle]):
* Source/WebKit/WebProcess/InjectedBundle/API/Cocoa/WKWebProcessPlugInNodeHandle.mm:
(protectedNodeHandle):
(-[WKWebProcessPlugInNodeHandle dealloc]):
(-[WKWebProcessPlugInNodeHandle htmlIFrameElementContentFrame]):
(-[WKWebProcessPlugInNodeHandle renderedImageWithOptions:width:]):
(-[WKWebProcessPlugInNodeHandle elementBounds]):
(-[WKWebProcessPlugInNodeHandle HTMLInputElementIsAutoFilled]):
(-[WKWebProcessPlugInNodeHandle HTMLInputElementIsAutoFilledAndViewable]):
(-[WKWebProcessPlugInNodeHandle HTMLInputElementIsAutoFilledAndObscured]):
(-[WKWebProcessPlugInNodeHandle setHTMLInputElementIsAutoFilled:]):
(-[WKWebProcessPlugInNodeHandle setHTMLInputElementIsAutoFilledAndViewable:]):
(-[WKWebProcessPlugInNodeHandle setHTMLInputElementIsAutoFilledAndObscured:]):
(-[WKWebProcessPlugInNodeHandle isHTMLInputElementAutoFillButtonEnabled]):
(-[WKWebProcessPlugInNodeHandle setHTMLInputElementAutoFillButtonEnabledWithButtonType:]):
(-[WKWebProcessPlugInNodeHandle htmlInputElementAutoFillButtonType]):
(-[WKWebProcessPlugInNodeHandle htmlInputElementLastAutoFillButtonType]):
(-[WKWebProcessPlugInNodeHandle HTMLInputElementIsUserEdited]):
(-[WKWebProcessPlugInNodeHandle HTMLTextAreaElementIsUserEdited]):
(-[WKWebProcessPlugInNodeHandle isSelectElement]):
(-[WKWebProcessPlugInNodeHandle isSelectableTextNode]):
(-[WKWebProcessPlugInNodeHandle isTextField]):
(-[WKWebProcessPlugInNodeHandle HTMLTableCellElementCellAbove]):
(-[WKWebProcessPlugInNodeHandle frame]):
* Source/WebKit/WebProcess/InjectedBundle/API/Cocoa/WKWebProcessPlugInRangeHandle.mm:
(protectedRangeHandle):
(-[WKWebProcessPlugInRangeHandle dealloc]):
(-[WKWebProcessPlugInRangeHandle frame]):
(-[WKWebProcessPlugInRangeHandle text]):
* Source/WebKit/WebProcess/InjectedBundle/API/Cocoa/WKWebProcessPlugInScriptWorld.mm:
(protectedWorld):
(-[WKWebProcessPlugInScriptWorld dealloc]):
(-[WKWebProcessPlugInScriptWorld clearWrappers]):
(-[WKWebProcessPlugInScriptWorld makeAllShadowRootsOpen]):
(-[WKWebProcessPlugInScriptWorld exposeClosedShadowRootsForExtensions]):
(-[WKWebProcessPlugInScriptWorld disableOverrideBuiltinsBehavior]):
* Source/WebKit/WebProcess/InjectedBundle/API/mac/WKDOMDocument.mm:
(-[WKDOMDocumentParserYieldToken initWithDocument:]):
* Source/WebKit/WebProcess/InjectedBundle/API/mac/WKDOMInternals.h:
* Source/WebKit/WebProcess/InjectedBundle/API/mac/WKDOMNode.mm:
(-[WKDOMNode _initWithImpl:]):
* Source/WebKit/WebProcess/InjectedBundle/API/mac/WKDOMRange.mm:
(-[WKDOMRange _initWithImpl:]):
* Source/WebKit/WebProcess/InjectedBundle/InjectedBundleHitTestResult.cpp:
(WebKit::InjectedBundleHitTestResult::nodeHandle const):
(WebKit::InjectedBundleHitTestResult::urlElementHandle const):
(WebKit::InjectedBundleHitTestResult::imageRect const):

Canonical link: https://commits.webkit.org/302475@main
